artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/JournalStorageManager.java

```diff
@@ -548,6 +549,17 @@ public LargeServerMessage createLargeMessage(final long id, final Message messag
 
          largeMessage.setMessageID(id);
 
+
+         // Check durable large massage size before to allocate resources if it can't be stored
+         if (largeMessage.isDurable()) {
+            final long maxRecordSize = getMaxRecordSize();
+            final int messageEncodeSize = largeMessage.getEncodeSize();
```

Comment:
You need to check the header and properties encoded size (that doesn't cost anything): this check will check size including header + trailer + body altogether.

Reply:
I see your point and I understand how to check the header and properties encoded size without costs from your example. In that point large message contains only header and properties of the source message and in addition it is precisely the record that is added to the journal. I think the last point is important because it let the check to work even if in the future other elements will be added to the large message.

Reply:
I see that you've put the checks before `validateFile` too, well done! :+1: